controllers/badges.js

```diff
@@ -0,0 +1,123 @@
+const badgeQuery = require("../models/badges");
+const { fetchUser } = require("../models/users");
+const imageService = require("../services/imageService");
+
+/**
+ * Get badges data
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ * @returns {Array} - Return badges
+ */
+const getBadges = async (req, res) => {
+  try {
+    const allBadges = await badgeQuery.fetchBadges(req.query);
+    return res.json({
+      message: "Badges returned successfully!",
+      badges: allBadges,
+    });
+  } catch (error) {
+    logger.error(`Error while fetching all badges: ${error}`);
+    return res.boom.serverUnavailable("Something went wrong please contact admin");
+  }
+};
+
+const getUserBadgeIds = async (req, res) => {
+  try {
+    const { userExists, badgeIds } = await badgeQuery.fetchUserBadgeIds(req.params.username);
+    let responseMsg = "";
+    if (userExists) {
+      responseMsg = badgeIds.length !== 0 ? "User badges returned successfully!" : "This user does not have any badges";
+      return res.json({ message: responseMsg, badgeIds });
+    } else {
+      return res.boom.notFound("The user does not exist");
+    }
+  } catch (error) {
+    logger.error(`Error while fetching all user badges: ${error}`);
+    return res.boom.serverUnavailable("Something went wrong please contact admin");
+  }
+};
```

Comment:
why do you need only badge id?

Reply:
My reasoning was to return badgeIds and map these badgeIds with 
response of `GET /badges` request i.e. badges.

Now I think it's better to return badges only for whom it is asked, 
as the ratio of visit of super-user: normal-user will be low compared to the other.